docs/architecture/adr-003.md

```diff
@@ -0,0 +1,69 @@
+# ADR 003: Add Fee Market Module
+
+## Changelog
+
+- 2022-01-06: first draft
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+This ADR proposes to add a new module "feemarket", allowing to define a common base fee for the network. This base fee is calculated dynamically in each block for the next block allowing to reflect on the activity of the network.
+
+## Context
+
+<!-- > This section describes the forces at play, including technological, political, social, and project local. These forces are probably in tension and should be called out as such. The language in this section is value-neutral. It is simply describing facts. It should clearly explain the problem and motivation that the proposal aims to resolve. -->
+
+Currently, the only way to enforce a transaction fee using Cosmos-SDK is to set up the local configuration "minimum-gas-prices" which has some limitations :
+
+- This value depend on the node local setting, which means each validators may use a different value which can be an issue for some webtools.
+- Some validators may not setup this value which would allow to spam the network with 0 fee txs.
+- To implement EIP-1559, a common base fee value is required for the network.
+
+## Decision
+
+<!-- > This section describes our response to these forces. It is stated in full sentences, with an active voice. "We will ..." -->
+
+The FeeMarket module has been introduced along with Ethermint v0.8.x which is required for supporting EIP-1559.
+
+In order to keep up with our efforts to stay compatible with Ethereum, we have decided to add this module along with the upgrade of ethermint in the next release.
+
+To have a better "pricing" mechanism when the network is congested, we have decided to keep the fee market fee's adjustment parameters close to what Ethereum uses. Those values can be tuned through governance vote if needed later on. 
+
+## Consequences
+
+<!-- > This section describes the resulting context, after applying the decision. All consequences should be listed here, not just the "positive" ones. A particular decision may have positive, negative, and neutral consequences, but all of them affect the team and project in the future. -->
+
+Requires a major network upgrade. 
+Local validator won't be able to choose arbitrary which fee value they would like to accept after the upgrade, but they will be able to define a local "tip" that can be added on top of this basefee. 
+
+### Backwards Compatibility
+
+<!-- All ADRs that introduce backward incompatibilities must include a section describing these incompatibilities and their severity. The ADR must explain how the author proposes to deal with these incompatibilities. ADR submissions without a sufficient backward compatibility treatise may be rejected outright. -->
+
+The proposed ADR is not backward compatible.
```

Comment:
I assume "legacy transactions" will still be accepted, or will they not? https://ethereum.stackexchange.com/questions/111654/how-legacy-transaction-are-handled-after-eip1559-update

Reply:
Yes you are right, I was thinking about cosmos transaction but at web3 level, it is backward compatible with legacy txs. 

I have fixed this section to add more clarity